pkg/relayer/session/claim.go

```diff
@@ -170,11 +162,20 @@ func (rs *relayerSessionsManager) waitForEarliestCreateClaimsHeight(
 	logger.Info().Msg("waiting & blocking until the earliest claim commit height for this supplier")
 
 	// Wait for the earliestSupplierClaimsCommitHeight to be reached before proceeding.
```

Comment:
Can you #PUC / decorate the logic here with a few more comments. 

It's dense but critical.